src/leader_election.rs

```diff
@@ -251,7 +273,14 @@ pub mod ballot_leader_election {
         }
 
         fn hb_timeout(&mut self) -> Option<Leader<Ballot>> {
+            info!(self.logger, "Heartbeat timeout");
+
             let result: Option<Leader<Ballot>> = if self.ballots.len() + 1 >= self.majority {
+                debug!(
+                    self.logger,
+                    "Majority confirmed, size: {}",
```

Comment:
```suggestion
                    "Received a majority of heartbeats, size: {}",
```

Log `self.ballots` instead of just the size.